quiche/src/lib.rs

```diff
@@ -9009,6 +9425,445 @@ mod tests {
         );
     }
 
+    #[test]
```

Comment:
Can we add some tests for contructing different types of bad server pereferred address too. I want to ensure that such things are caught as early as possible and that any future changes can be caught in a regression test before they hit apps.

Reply:
Yes. I refactored the existing `server_preferred_address_rejects_zero_len_cid()` test and created a new one called `server_preferred_address_rejects_empty_v4_and_empty_v6_preferred_addresses()`